mods/maintain-flux-colour-temperature.wh.cpp

```diff
@@ -0,0 +1,349 @@
+// @author          Roland Pihlakas
+// @github          https://github.com/levitation-opensource/
+// @homepage        https://www.simplify.ee/
+// @compilerOptions -lkernel32
+// @include         flux.exe
+// ==/WindhawkMod==
+
+// Source code is published under The GNU General Public License v3.0.
+//
+// For bug reports and feature requests, please open an issue here:
+// https://github.com/levitation-opensource/my-windhawk-mods/issues
+//
+// For pull requests, development takes place here:
+// https://github.com/levitation-opensource/my-windhawk-mods/
+
+// ==WindhawkModReadme==
+/*
+# Maintain colour temperature of f.lux
+
+Keeps your preferred f.lux colour temperature settings, eliminating automatic changes. You can choose the desired colour temperature from f.lux dropdown menu, and f.lux will not change it later automatically.
+
+The mod operates by providing f.lux always with a fixed day and time. This fixed day and time can be adjusted by the user in the settings panel of this mod, if necessary, though the default setting should be usually suitable. By default the noon of approximately midsummer day of Northern Hemisphere is chosen as the fixed day and time. The particular day and time are not very important, what matters is that the fixed time is not near the time points where f.lux would usually get triggered to adjust the screen colour on its own.
+
+There are many posts in f.lux forum asking for this functionality since 2016. Unfortunately there were no solutions provided so far:
+* [Disable f.lux's automatic adjustments based on time/location](https://forum.justgetflux.com/topic/3012/disable-f-lux-s-automatic-adjustments-based-on-time-location/)
+* [Can automatic timer settings be disabled?](https://forum.justgetflux.com/topic/6162/can-automatic-timer-settings-be-disabled)
+* [Disabling all automatic functions.](https://forum.justgetflux.com/topic/6228/disabling-all-automatic-functions)
+* [How To Disable Automatic Change Based On Time / Location ?](https://forum.justgetflux.com/topic/7904/how-to-disable-automatic-change-based-on-time-location)
+* [How do I stop flux to stop changing colors arbitrary / Force always on mode / Force coupling day and night colors?](https://forum.justgetflux.com/topic/8404/how-do-i-stop-flux-to-stop-changing-colors-arbitrary-force-always-on-mode-force-coupling-day-and-night-colors)
+* [Sleeping during the day](https://forum.justgetflux.com/topic/8431/sleeping-during-the-day)
+* [How do I make my own Permanent preset?](https://forum.justgetflux.com/topic/8475/how-do-i-make-my-own-permanent-preset)
+* [Disable Circadian Response Only](https://forum.justgetflux.com/topic/8502/disable-circadian-response-only/)
+*/
+// ==/WindhawkModReadme==
+
+// ==WindhawkModSettings==
+/*
+- Hour: 12
+  $name: Hour in local time
+- Day: 21
+  $name: Day
+- Month: 6
+  $name: Month
+*/
+// ==/WindhawkModSettings==
+
+
+#include <windowsx.h>
+
+
+#ifndef WH_MOD
+#define WH_MOD
+#include <mods_api.h>
+#endif

```

Comment:
Why was this added?

Reply:
I am using Visual Studio to write my code. This enables intellisense checks and other conveniences. Also it enables me saving the code while it is still incomplete and not ready to compile. 

This header is included for intellisense code analysis purposes. Else I get lots of noisy error messages from intellisense. And of course it also means that I am unable to see real errors while I am developing since they are drowned in the noise. 

Including these lines does no harm since they activate only during development in editors external to Windhawk.

Reply:
You're free to use it during development, I just think that adding it can be confusing for readers of the mod code. `WH_MOD` is an internal definition of Windhawk, and its meaning may change in the future, or it can be removed. I see little risk here since even if removed, this block won't have any effect, I just think that it shouldn't be part of the submitted mod.

Reply:
Ok. I removed the WH_MOD usage and mods_api.h inclusion. Maybe there is some other way to achieve the compatibility with intellisense?

Reply:
Depending on your environment, perhaps you can configure "Forced Include File":
https://learn.microsoft.com/en-us/cpp/build/reference/fi-name-forced-include-file?view=msvc-170